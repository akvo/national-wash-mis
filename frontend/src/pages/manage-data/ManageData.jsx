import React, { useState, useEffect } from "react";
import "./style.scss";
import {
  Row,
  Col,
  Card,
  Button,
  Breadcrumb,
  Divider,
  Typography,
  Table,
  message,
  ConfigProvider,
  Empty,
} from "antd";
import { Link } from "react-router-dom";
import { PlusSquareOutlined, CloseSquareOutlined } from "@ant-design/icons";
import { api } from "../../lib";
import { DataFilters } from "../../components";

const { Title } = Typography;

const renderDetails = (record) => {
  return (
    <div>
      <div className="expand-wrap">Details View</div>
      <div className="expand-footer">
        <div>
          <Link to={"/user/edit/" + record.id}>
            <Button type="secondary">Delete</Button>
          </Link>{" "}
          <Button danger>Upload CSV</Button>
        </div>
      </div>
    </div>
  );
};

const ManageData = () => {
  const [loading, setLoading] = useState(false);
  const [initializing, setInitializing] = useState(true);
  const [dataset, setDataset] = useState([]);
  const [form, setForm] = useState(null);
  const [forms, setForms] = useState([]);
  const [query, setQuery] = useState("");

  const columns = [
    {
      title: "Name",
      dataIndex: "name",
      key: "name",
    },
    {
      title: "Last Updated",
      dataIndex: "updated_at",
      render: () => "Saturday 17 November 2021",
    },
    {
      title: "User",
      dataIndex: "user",
      render: () => "Ouma Odhiambo",
    },
    {
      title: "Region",
      dataIndex: "administration",
      render: (administration) => administration?.name || "-",
    },
    Table.EXPAND_COLUMN,
  ];

  const handleChange = () => {
    // setCurrentPage(e.current);
  };

  useEffect(() => {
    setInitializing(true);
    api
      .get("forms/")
      .then((res) => {
        setForms(res.data);
        setInitializing(false);
      })
      .catch((err) => {
        message.error("Could not load forms");
        setInitializing(false);
        console.error(err);
      });
  }, []);

  useEffect(() => {
    if (form) {
      setLoading(true);
      api
        .get(`web/form/${form}/`)
        .then((res) => {
          setDataset(res.data);
          setLoading(false);
        })
        .catch((err) => {
          message.error("Could not load forms");
          setLoading(false);
          console.error(err);
        });
    }
  }, [form]);

  return (
    <div id="manageData">
      <Row justify="space-between">
        <Col>
          <Breadcrumb
            separator={
              <h2 className="ant-typography" style={{ display: "inline" }}>
                {">"}
              </h2>
            }
          >
            <Breadcrumb.Item>
              <Link to="/control-center">
                <Title style={{ display: "inline" }} level={2}>
                  Control Center
                </Title>
              </Link>
            </Breadcrumb.Item>
            <Breadcrumb.Item>
              <Title style={{ display: "inline" }} level={2}>
                Manage Data
              </Title>
            </Breadcrumb.Item>
          </Breadcrumb>
        </Col>
      </Row>
      <Divider />
      <DataFilters
        query={query}
        setQuery={setQuery}
        form={form}
        setForm={setForm}
        forms={forms}
        loading={loading || initializing}
      />
      <Divider />
      <Card
        style={{ padding: 0, minHeight: "40vh" }}
        bodyStyle={{ padding: 0 }}
      >
        <ConfigProvider
          renderEmpty={() => (
            <Empty description={form ? "No data" : "No form selected"} />
          )}
        >
          <Table
            columns={columns}
            dataSource={dataset.question_group}
            loading={loading || initializing}
            onChange={handleChange}
            // pagination={{
            //   total: totalCount,
            //   pageSize: 10,
            // }}
            rowKey="name"
            expandable={{
              expandedRowRender: renderDetails,
              expandIcon: ({ expanded, onExpand, record }) =>
                expanded ? (
                  <CloseSquareOutlined
                    onClick={(e) => onExpand(record, e)}
                    style={{ color: "#e94b4c" }}
                  />
                ) : (
                  <PlusSquareOutlined
                    onClick={(e) => onExpand(record, e)}
                    style={{ color: "#7d7d7d" }}
                  />
                ),
            }}
          />
        </ConfigProvider>
      </Card>
    </div>
  );
};

export default React.memo(ManageData);
