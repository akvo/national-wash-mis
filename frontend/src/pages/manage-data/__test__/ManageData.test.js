import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import ManageData from "../ManageData";
import { MemoryRouter } from "react-router-dom";
import axios from "axios";

window.matchMedia =
  window.matchMedia ||
  function () {
    return {
      matches: false,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    };
  };

jest.mock("axios");

describe("page/manage-data", () => {
  test("test if ManageData has a list of data", async () => {
    const { container, getByText, getByTestId } = render(<ManageData />, {
      wrapper: MemoryRouter,
    });
    expect(screen.getByText(/Control Center/i)).toBeInTheDocument();

    const fakeData = [
      {
        id: 2152,
        name: "Churo/Amaya - Howard Inc",
        form: 974754029,
        administration: "Churo/Amaya",
        geo: [36.441970658358, 0.868986472010182],
        created_by: "Turkana North Approver",
        updated_by: null,
        created: "January 31, 2022",
        updated: null,
        pending_data: null,
      },
      {
        id: 2156,
        name: "Tangulbei/Korossi - Smith, Robles and Spencer",
        form: 974754029,
        administration: "Tangulbei/Korossi",
        geo: [36.1880630895679, 0.7986885954899671],
        created_by: "unknown 1 User",
        updated_by: null,
        created: "December 23, 2021",
        updated: null,
        pending_data: null,
      },
    ];

    axios.mockResolvedValueOnce({ status: 200, data: fakeData });

    const tableBody = container.querySelector(".ant-table-tbody");
    expect(tableBody.firstChild).toHaveClass("ant-table-placeholder");

    const select = container.querySelector(
      ".ant-select-selection-search-input"
    );
    fireEvent.change(select, { target: { value: "CLT" } });

    const result = container.querySelector(".ant-select-selection-item");
    expect(result).toBeDefined();

    fireEvent.click(getByText("Remove Filters"));
    expect(container.querySelector(".ant-select-selection-item")).toBeNull();

    fireEvent.click(getByTestId("advanced-filter"));
    const advancedFilter = container.querySelector(".advanced-filters");
    expect(advancedFilter).toBeDefined();

    expect(container).toMatchSnapshot();
  });
});
