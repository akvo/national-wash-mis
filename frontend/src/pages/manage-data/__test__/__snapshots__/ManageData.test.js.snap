// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page/manage-data test if ManageData has a list of data 1`] = `
<div>
  <div
    id="manageData"
  >
    <div
      class="ant-row ant-row-space-between"
    >
      <div
        class="ant-col"
      >
        <div
          class="ant-breadcrumb"
        >
          <span>
            <span
              class="ant-breadcrumb-link"
            >
              <a
                href="/control-center"
              >
                <h1
                  class="ant-typography"
                  style="display: inline;"
                >
                  Control Center
                </h1>
              </a>
            </span>
            <span
              class="ant-breadcrumb-separator"
            >
              &gt;
            </span>
          </span>
          <span>
            <span
              class="ant-breadcrumb-link"
            >
              <h2
                class="ant-typography"
                style="display: inline;"
              >
                Manage Data
              </h2>
            </span>
            <span
              class="ant-breadcrumb-separator"
            >
              &gt;
            </span>
          </span>
        </div>
        <div
          class="description-paragraph"
        >
          This is where you :
          <ul>
            <li>
              Add new data using webforms
            </li>
            <li>
              Bulk upload data using spreadsheets
            </li>
            <li>
              Download data
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="ant-tabs ant-tabs-top ant-tabs-large"
    >
      <div
        class="ant-tabs-nav"
        role="tablist"
      >
        <div
          class="ant-tabs-nav-wrap"
        >
          <div
            class="ant-tabs-nav-list"
            style="transform: translate(0px, 0px);"
          >
            <div
              class="ant-tabs-tab"
            >
              <div
                aria-controls="rc-tabs-test-panel-/data/manage"
                aria-selected="false"
                class="ant-tabs-tab-btn"
                id="rc-tabs-test-tab-/data/manage"
                role="tab"
                tabindex="0"
              >
                Manage Data
              </div>
            </div>
            <div
              class="ant-tabs-tab"
            >
              <div
                aria-controls="rc-tabs-test-panel-/data/export"
                aria-selected="false"
                class="ant-tabs-tab-btn"
                id="rc-tabs-test-tab-/data/export"
                role="tab"
                tabindex="0"
              >
                Downloaded Data
              </div>
            </div>
            <div
              class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
            />
          </div>
        </div>
        <div
          class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
        >
          <button
            aria-controls="rc-tabs-test-more-popup"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-hidden="true"
            class="ant-tabs-nav-more"
            id="rc-tabs-test-more"
            style="visibility: hidden; order: 1;"
            tabindex="-1"
            type="button"
          >
            <span
              aria-label="ellipsis"
              class="anticon anticon-ellipsis"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="ellipsis"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        class="ant-tabs-content-holder"
      >
        <div
          class="ant-tabs-content ant-tabs-content-top"
        >
          <div
            aria-hidden="true"
            aria-labelledby="rc-tabs-test-tab-/data/manage"
            class="ant-tabs-tabpane"
            id="rc-tabs-test-panel-/data/manage"
            role="tabpanel"
            style="display: none;"
            tabindex="-1"
          />
          <div
            aria-hidden="true"
            aria-labelledby="rc-tabs-test-tab-/data/export"
            class="ant-tabs-tabpane"
            id="rc-tabs-test-panel-/data/export"
            role="tabpanel"
            style="display: none;"
            tabindex="-1"
          />
        </div>
      </div>
    </div>
    <div
      class="ant-row"
    >
      <div
        class="ant-col"
        style="flex: 1 1 auto;"
      >
        <div
          class="ant-space ant-space-horizontal ant-space-align-center"
        >
          <div
            class="ant-space-item"
            style="margin-right: 8px;"
          >
            <div
              class="ant-select form-dropdown  ant-select-single ant-select-show-arrow ant-select-open"
              style="width: 160px;"
            >
              <div
                class="ant-select-selector"
              >
                <span
                  class="ant-select-selection-search"
                >
                  <input
                    aria-activedescendant="rc_select_TEST_OR_SSR_list_-1"
                    aria-autocomplete="list"
                    aria-controls="rc_select_TEST_OR_SSR_list"
                    aria-expanded="true"
                    aria-haspopup="listbox"
                    aria-owns="rc_select_TEST_OR_SSR_list"
                    autocomplete="off"
                    class="ant-select-selection-search-input"
                    id="rc_select_TEST_OR_SSR"
                    readonly=""
                    role="combobox"
                    style="opacity: 0;"
                    type="search"
                    unselectable="on"
                    value=""
                  />
                </span>
                <span
                  class="ant-select-selection-placeholder"
                  style="visibility: hidden;"
                >
                  Select Form
                </span>
              </div>
              <span
                aria-hidden="true"
                class="ant-select-arrow"
                style="user-select: none;"
                unselectable="on"
              >
                <span
                  aria-label="down"
                  class="anticon anticon-down ant-select-suffix"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </span>
              </span>
              <div
                style="position: absolute; top: 0px; left: 0px; width: 100%;"
              >
                <div>
                  <div
                    class="ant-select-dropdown ant-select-dropdown-empty"
                    style="opacity: 0; pointer-events: none; min-width: 0; width: 0px;"
                  >
                    <div>
                      <div
                        class="ant-select-item-empty"
                        id="rc_select_TEST_OR_SSR_list"
                        role="listbox"
                      >
                        <div
                          class="ant-empty ant-empty-normal ant-empty-small"
                        >
                          <div
                            class="ant-empty-image"
                          >
                            <svg
                              class="ant-empty-img-simple"
                              height="41"
                              viewBox="0 0 64 41"
                              width="64"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <g
                                fill="none"
                                fill-rule="evenodd"
                                transform="translate(0 1)"
                              >
                                <ellipse
                                  class="ant-empty-img-simple-ellipse"
                                  cx="32"
                                  cy="33"
                                  rx="32"
                                  ry="7"
                                />
                                <g
                                  class="ant-empty-img-simple-g"
                                  fill-rule="nonzero"
                                >
                                  <path
                                    d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                                  />
                                  <path
                                    class="ant-empty-img-simple-path"
                                    d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                                  />
                                </g>
                              </g>
                            </svg>
                          </div>
                          <div
                            class="ant-empty-description"
                          >
                            No Data
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-space-item"
            style="margin-right: 8px;"
          >
            <div
              class="ant-space ant-space-horizontal ant-space-align-center"
            >
              <div
                class="ant-space-item"
              >
                <div>
                  
                  <div
                    class="ant-select ant-select-single ant-select-allow-clear ant-select-show-arrow ant-select-show-search"
                    style="width: 160px;"
                  >
                    <div
                      class="ant-select-selector"
                    >
                      <span
                        class="ant-select-selection-search"
                      >
                        <input
                          aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                          aria-autocomplete="list"
                          aria-controls="rc_select_TEST_OR_SSR_list"
                          aria-haspopup="listbox"
                          aria-owns="rc_select_TEST_OR_SSR_list"
                          autocomplete="off"
                          class="ant-select-selection-search-input"
                          id="rc_select_TEST_OR_SSR"
                          role="combobox"
                          type="search"
                          value=""
                        />
                      </span>
                      <span
                        class="ant-select-selection-placeholder"
                      >
                        Select County
                      </span>
                    </div>
                    <span
                      aria-hidden="true"
                      class="ant-select-arrow"
                      style="user-select: none;"
                      unselectable="on"
                    >
                      <span
                        aria-label="down"
                        class="anticon anticon-down ant-select-suffix"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="down"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                          />
                        </svg>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-space-item"
            style="margin-right: 8px;"
          >
            <button
              ant-click-animating-without-extra-node="false"
              class="ant-btn light"
              type="button"
            >
              <span>
                Remove Filters
              </span>
            </button>
          </div>
          <div
            class="ant-space-item"
          >
            <button
              ant-click-animating-without-extra-node="false"
              class="ant-btn light active"
              data-testid="advanced-filter"
              type="button"
            >
              <span
                aria-label="filter"
                class="anticon anticon-filter"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="filter"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M880.1 154H143.9c-24.5 0-39.8 26.7-27.5 48L349 597.4V838c0 17.7 14.2 32 31.8 32h262.4c17.6 0 31.8-14.3 31.8-32V597.4L907.7 202c12.2-21.3-3.1-48-27.6-48zM603.4 798H420.6V642h182.9v156zm9.6-236.6l-9.5 16.6h-183l-9.5-16.6L212.7 226h598.6L613 561.4z"
                  />
                </svg>
              </span>
              <span>
                Advanced Filters
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="advanced-filters"
    >
      <div
        class="ant-card ant-card-bordered"
        style="padding: 0px;"
      >
        <div
          class="ant-card-body"
          style="padding: 12px;"
        >
          <div>
            <div
              class="ant-select ant-select-single ant-select-show-arrow ant-select-show-search"
              style="width: 100%;"
            >
              <div
                class="ant-select-selector"
              >
                <span
                  class="ant-select-selection-search"
                >
                  <input
                    aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                    aria-autocomplete="list"
                    aria-controls="rc_select_TEST_OR_SSR_list"
                    aria-haspopup="listbox"
                    aria-owns="rc_select_TEST_OR_SSR_list"
                    autocomplete="off"
                    class="ant-select-selection-search-input"
                    id="rc_select_TEST_OR_SSR"
                    role="combobox"
                    type="search"
                    value=""
                  />
                </span>
                <span
                  class="ant-select-selection-placeholder"
                >
                  Search..
                </span>
              </div>
              <span
                aria-hidden="true"
                class="ant-select-arrow"
                style="user-select: none;"
                unselectable="on"
              >
                <span
                  aria-label="down"
                  class="anticon anticon-down ant-select-suffix"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-divider ant-divider-horizontal"
      role="separator"
    />
    <div
      class="ant-card ant-card-bordered"
      style="padding: 0px; min-height: 40vh;"
    >
      <div
        class="ant-card-body"
        style="padding: 0px;"
      >
        <div
          class="ant-table-wrapper"
        >
          <div
            class="ant-spin-nested-loading"
          >
            <div
              class="ant-spin-container"
            >
              <div
                class="ant-table ant-table-empty"
              >
                <div
                  class="ant-table-container"
                >
                  <div
                    class="ant-table-content"
                  >
                    <table
                      style="table-layout: auto;"
                    >
                      <colgroup>
                        <col />
                        <col />
                        <col />
                        <col />
                        <col
                          class="ant-table-expand-icon-col"
                        />
                      </colgroup>
                      <thead
                        class="ant-table-thead"
                      >
                        <tr>
                          <th
                            class="ant-table-cell"
                          >
                            Name
                          </th>
                          <th
                            class="ant-table-cell"
                          >
                            Last Updated
                          </th>
                          <th
                            class="ant-table-cell"
                          >
                            User
                          </th>
                          <th
                            class="ant-table-cell"
                          >
                            Region
                          </th>
                          <th
                            class="ant-table-cell ant-table-row-expand-icon-cell"
                          >
                            
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        class="ant-table-tbody"
                      >
                        <tr
                          class="ant-table-placeholder"
                        >
                          <td
                            class="ant-table-cell"
                            colspan="5"
                          >
                            <div
                              class="ant-empty"
                            >
                              <div
                                class="ant-empty-image"
                              >
                                <svg
                                  class="ant-empty-img-default"
                                  height="152"
                                  viewBox="0 0 184 152"
                                  width="184"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <g
                                    fill="none"
                                    fill-rule="evenodd"
                                  >
                                    <g
                                      transform="translate(24 31.67)"
                                    >
                                      <ellipse
                                        class="ant-empty-img-default-ellipse"
                                        cx="67.797"
                                        cy="106.89"
                                        rx="67.797"
                                        ry="12.668"
                                      />
                                      <path
                                        class="ant-empty-img-default-path-1"
                                        d="M122.034 69.674L98.109 40.229c-1.148-1.386-2.826-2.225-4.593-2.225h-51.44c-1.766 0-3.444.839-4.592 2.225L13.56 69.674v15.383h108.475V69.674z"
                                      />
                                      <path
                                        class="ant-empty-img-default-path-2"
                                        d="M101.537 86.214L80.63 61.102c-1.001-1.207-2.507-1.867-4.048-1.867H31.724c-1.54 0-3.047.66-4.048 1.867L6.769 86.214v13.792h94.768V86.214z"
                                        transform="translate(13.56)"
                                      />
                                      <path
                                        class="ant-empty-img-default-path-3"
                                        d="M33.83 0h67.933a4 4 0 0 1 4 4v93.344a4 4 0 0 1-4 4H33.83a4 4 0 0 1-4-4V4a4 4 0 0 1 4-4z"
                                      />
                                      <path
                                        class="ant-empty-img-default-path-4"
                                        d="M42.678 9.953h50.237a2 2 0 0 1 2 2V36.91a2 2 0 0 1-2 2H42.678a2 2 0 0 1-2-2V11.953a2 2 0 0 1 2-2zM42.94 49.767h49.713a2.262 2.262 0 1 1 0 4.524H42.94a2.262 2.262 0 0 1 0-4.524zM42.94 61.53h49.713a2.262 2.262 0 1 1 0 4.525H42.94a2.262 2.262 0 0 1 0-4.525zM121.813 105.032c-.775 3.071-3.497 5.36-6.735 5.36H20.515c-3.238 0-5.96-2.29-6.734-5.36a7.309 7.309 0 0 1-.222-1.79V69.675h26.318c2.907 0 5.25 2.448 5.25 5.42v.04c0 2.971 2.37 5.37 5.277 5.37h34.785c2.907 0 5.277-2.421 5.277-5.393V75.1c0-2.972 2.343-5.426 5.25-5.426h26.318v33.569c0 .617-.077 1.216-.221 1.789z"
                                      />
                                    </g>
                                    <path
                                      class="ant-empty-img-default-path-5"
                                      d="M149.121 33.292l-6.83 2.65a1 1 0 0 1-1.317-1.23l1.937-6.207c-2.589-2.944-4.109-6.534-4.109-10.408C138.802 8.102 148.92 0 161.402 0 173.881 0 184 8.102 184 18.097c0 9.995-10.118 18.097-22.599 18.097-4.528 0-8.744-1.066-12.28-2.902z"
                                    />
                                    <g
                                      class="ant-empty-img-default-g"
                                      transform="translate(149.65 15.383)"
                                    >
                                      <ellipse
                                        cx="20.654"
                                        cy="3.167"
                                        rx="2.849"
                                        ry="2.815"
                                      />
                                      <path
                                        d="M5.698 5.63H0L2.898.704zM9.259.704h4.985V5.63H9.259z"
                                      />
                                    </g>
                                  </g>
                                </svg>
                              </div>
                              <div
                                class="ant-empty-description"
                              >
                                No form selected
                              </div>
                            </div>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
